2015-Fall/lrusnac/Calculator/Calculator/ViewController.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by Leonid Rusnac on 01/09/15.
+//  Copyright (c) 2015 Leonid Rusnac. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    @IBOutlet weak var display: UILabel!
+
+    var userIsInTheMiddleOfTypingANumber = false
+    var pointWasInsertedInThisNumber = false
+    var lastValue = 0.0
+    var lastOperation: String?
+    
+    @IBAction func doOperation(sender: UIButton) {
+        let operation = sender.currentTitle!
+        if lastOperation != nil {
+            evaluate()
+        } else {
+            lastValue = NSNumberFormatter().numberFromString(display.text!)!.doubleValue
+        }
+        
+        userIsInTheMiddleOfTypingANumber = false
+        pointWasInsertedInThisNumber = false
+        lastOperation = operation
+    }
+    
+    @IBAction func clear() {
+        lastValue = 0.0
+        lastOperation = nil
+        display.text = "0"
+        userIsInTheMiddleOfTypingANumber = false
+        pointWasInsertedInThisNumber = false
+    }
+    
+    @IBAction func appendDigit(sender: UIButton) {
+        let digit = sender.currentTitle!
+        if userIsInTheMiddleOfTypingANumber {
+            display.text = display.text! + digit
+        } else {
+            display.text = digit
+            userIsInTheMiddleOfTypingANumber = true
+        }
+    }
+    
+    @IBAction func addPoind() {
+        if !userIsInTheMiddleOfTypingANumber {
+            userIsInTheMiddleOfTypingANumber = true
+            display.text = "0."
+        } else {
+            if !pointWasInsertedInThisNumber {
+                display.text = display.text! + "."
+            }
+        }
+        pointWasInsertedInThisNumber = true
+    }
+    
+    func evaluate() {
+        let displayValue = NSNumberFormatter().numberFromString(display.text!)!.doubleValue //found on the internet, must be a better way
```

Comment:
Albeit a bit longish, I think this is the way to do it for now in Swift. Maybe something shorter will come around. By casting to objective-c you could go `(display.text! as NSString).doubleValue` but I would prefer to solve it in pure Swift. If anything, you can make your code a but more robust here by unwrapping the optionals with `if let` - which we did not cover today.